torch/csrc/jit/codegen/cuda/lower_insert_syncs.cpp

```diff
@@ -419,6 +520,8 @@ class ReadAfterWriteSyncs : public kir::MutableIrVisitor {
  private:
   //! Keep track of expressions that must be followed by syncthreads
   std::deque<kir::Expr*> sync_after_;
+  //! Keep track of write expressions that must be placed before syncthreads
```

Comment:
I think this tracks the last write to the shared memory buffer we're trying to sync? Comment is a little unclear had to gather what's going on from context.

Reply:
Yes. Added more comments.